full-service/src/json_rpc/v2/e2e_tests/transaction/build_submit/build_and_submit_with_spend_from_subaddress.rs

```diff
@@ -307,4 +308,182 @@ mod e2e_transaction {
+
+        // Add an account
+        let body = json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "create_account",
+            "params": {
+                "name": "Exchange Main Account",
+                "require_spend_subaddresses": true,
+            }
+        });
+        let res = dispatch(&client, body, &logger);
+        let result = res.get("result").unwrap();
+        let account_obj = result.get("account").unwrap();
+        let account_id = account_obj.get("id").unwrap().as_str().unwrap();
+
+        let (
+            (alice_public_address, alice_b58_public_address),
+            (bob_public_address, bob_b58_public_address),
+            (carol_public_address, carol_b58_public_address),
+        ) = [
+            "Subaddress for Alice",
+            "Subaddress for Bob",
+            "Subaddress for Carol",
+        ]
+        .iter()
+        .map(|metadata| {
+            let body = json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "assign_address_for_account",
+            "params": {
+            "account_id": account_id,
+            "metadata": metadata,
+            }
+            });
+            let res = dispatch(&client, body, &logger);
+            let result = res.get("result").unwrap();
+            let address = result.get("address").unwrap();
+            let b58_address = address.get("public_address_b58").unwrap().as_str().unwrap();
+            let public_address = b58_decode_public_address(b58_address).unwrap();
+            (public_address, b58_address.to_string())
+        })
+        .collect_tuple()
+        .unwrap();
+
+        // Add a block with a txo for Bob
+        add_block_to_ledger_db(
+            &mut ledger_db,
+            &vec![bob_public_address.clone()],
+            200 * MOB,
+            &[KeyImage::from(rng.next_u64())],
+            &mut rng,
+        );
+
+        manually_sync_account(
+            &ledger_db,
+            &db_ctx.get_db_instance(logger.clone()),
+            &AccountID(account_id.to_string()),
+            &logger,
+        );
+
+        // Imagine that Bob is sending 42.0 MOB to Alice
+        let body = json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "build_and_submit_transaction",
+            "params": {
+                "account_id": account_id,
+                "recipient_public_address": alice_b58_public_address,
+                "amount": { "value": "42000000000000", "token_id": "0" }, // 42.0 MOB
+            }
+        });
+        let res = dispatch(&client, body, &logger);
+        let error = res.get("error").unwrap();
+        let data = error.get("data").unwrap();
+        let details = data.get("details").unwrap();
+        assert!(details
+            .to_string()
+            .contains(&transaction_error("This account is configured to spend only from a specific subaddress. Please provide a subaddress to spend from.".to_string()).to_string()));
```

Comment:
tl;dr Nothing actionable here, just me capturing thoughts from the squirrel that runs around in my skull

I think it's good that this is testing the error condition as the error cases often get skipped in testing and they're the ones that may have the most impact on the end user.
Saying all that, checking for a large substring feels like a future pain point in that anytime someone wants to update the error message they have to update the test. Maybe that's a good thing since in many ways this error message is the public API to the user.

Reply:
Agreed - I think we could use an error overhaul - @holtzman and I were talking about how that might take shape. Will leave as future work.